oak-core/src/main/java/org/apache/jackrabbit/oak/query/xpath/Statement.java

```diff
@@ -353,29 +355,23 @@ private static void appendQueryOptions(StringBuilder buff, QueryOptions queryOpt
             return;
         }
         buff.append(" option(");
-        int optionCount = 0;
+        List<String> optionValues = new ArrayList<>();
         if (queryOptions.traversal != Traversal.DEFAULT) {
-            buff.append("traversal " + queryOptions.traversal);
-            optionCount++;
+            optionValues.add("traversal " + queryOptions.traversal);
         }
         if (queryOptions.indexName != null) {
-            if (optionCount > 0) {
-                buff.append(", ");
-            }
-            buff.append("index name [");
-            buff.append(queryOptions.indexName);
-            buff.append("]");
-            optionCount++;
+            optionValues.add("index name [" + queryOptions.indexName + "]");
         }
         if (queryOptions.indexTag != null) {
-            if (optionCount > 0) {
-                buff.append(", ");
-            }
-            buff.append("index tag [");
-            buff.append(queryOptions.indexTag);
-            buff.append("]");
-            optionCount++;
+            optionValues.add("index tag [" + queryOptions.indexTag + "]");
+        }
+        if (queryOptions.offset != -1) {
+            optionValues.add("offset " + queryOptions.offset);
+        }
+        if (queryOptions.limit != -1) {
+            optionValues.add("limit " + queryOptions.limit);
         }
+        buff.append(optionValues.stream().collect(Collectors.joining(", ")));
```

Comment:
this can be simplified with 
```suggestion
        buff.append(String.join(", ", optionValues));
```